### PR TITLE
Added SPV sendToAddress RPC

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds SPV sendToAddress RPC

#### Which issue(s) does this PR fixes?:

Fixes part of #202

### CODE REVIEW

**Review by fuxingloh — APPROVED** [review]

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 1]
```diff
@@ -39,6 +42,19 @@ export class Spv {
...
+  /**
+   * Send a Bitcoin amount to a given address.
+   *
+   * @param {string} address Bitcoin address
+   * @param {number} amount Bitcoin amount
+   * @param {SendToAddressOptions} [options]
+   * @param {number} [options.feerate] Fee rate in satoshis per KB. Minimum is 1000. Defaults to 1200. See DEFAULT_FEE_RATE_IN_SATOSHIS
+   * @return {Promise<SendMessageResult>}
+   */
+  async sendToAddress (address: string, amount: number, options = { feerate: DEFAULT_FEE_RATE_IN_SATOSHIS }): Promise<SendMessageResult> {
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/spv.ts:55]:
```suggestion
  async sendToAddress (address: string, amount: BigNumber, options = { feerate: DEFAULT_FEE_RATE_IN_SATOSHIS }): Promise<SendMessageResult> {
```

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import { ApiClient } from '../.'
 
+export const DEFAULT_FEE_RATE = 0.000012
+export const DEFAULT_FEE_RATE_IN_SATOSHIS = 1200
```
Comment (fuxingloh) [inline packages/jellyfish-api-core/src/category/spv.ts:4]: use BigNumber

**Review by fuxingloh — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import { ApiClient } from '../.'
+import BigNumber from 'bignumber.js'
+
+export const DEFAULT_FEE_RATE_IN_SATOSHIS = new BigNumber('10000')
```
Comment (canonbrother) [inline packages/jellyfish-api-core/src/category/spv.ts:4]:
```suggestion
export const DEFAULT_FEE_RATE_IN_SATOSHI = new BigNumber('10000')
```

File: website/docs/jellyfish/api/spv.md [hunk 4]
```diff
@@ -50,3 +50,22 @@ interface ReceivedByAddressInfo {
...
 }
 ```
+
+## sendToAddress
+
+Send a Bitcoin amount to a given address.
+
+```ts title="client.spv.sendToAddress()"
+interface spv {
+  sendToAddress (address: string, amount: BigNumber, options: SendToAddressOptions = { feerate: 10000 }): Promise<SendMessageResult>
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/spv.md:60]: `sendToAddress (address: string, amount: number, options = { feerate: DEFAULT_FEE_RATE_IN_SATOSHI }): Promise<SendMessageResult>`

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 5]
```diff
@@ -39,6 +42,19 @@ export class Spv {
...
+  /**
+   * Send a Bitcoin amount to a given address.
+   *
+   * @param {string} address Bitcoin address
+   * @param {BigNumber} amount Bitcoin amount
+   * @param {SendToAddressOptions} [options]
+   * @param {BigNumber} [options.feerate] Fee rate in satoshis per KB. Minimum is 1000. Defaults to 10000. See DEFAULT_FEE_RATE_IN_SATOSHIS
+   * @return {Promise<SendMessageResult>}
+   */
+  async sendToAddress (address: string, amount: BigNumber, options = { feerate: DEFAULT_FEE_RATE_IN_SATOSHIS }): Promise<SendMessageResult> {
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/spv.ts:55]: DEFAULT_FEE_RATE_IN_SATOSHI

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 6]
```diff
@@ -39,6 +42,19 @@ export class Spv {
...
     return await this.client.call('spv_listreceivedbyaddress', [minConfirmation, address], 'number')
   }
+
+  /**
+   * Send a Bitcoin amount to a given address.
+   *
+   * @param {string} address Bitcoin address
+   * @param {BigNumber} amount Bitcoin amount
+   * @param {SendToAddressOptions} [options]
+   * @param {BigNumber} [options.feerate] Fee rate in satoshis per KB. Minimum is 1000. Defaults to 10000. See DEFAULT_FEE_RATE_IN_SATOSHIS
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/spv.ts:52]: satoshi

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/spv.ts:52]: DEFAULT_FEE_RATE_IN_SATOSHI

File: packages/jellyfish-api-core/src/category/spv.ts [hunk 7]
```diff
@@ -53,3 +69,12 @@ export interface ReceivedByAddressInfo {
   /** The ids of transactions received by the address */
   txids: string[]
 }
+
+export interface SendToAddressOptions {
+  feerate: BigNumber
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/spv.ts:74]: I take a quick look. 
Feel that shouldn't this 
`feerate?: BigNumber`

Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/spv.ts:74]: Is there an English word = `feerate`? If not shouldn't this be `feeRate`

File: packages/jellyfish-api-core/src/category/spv.ts (hunk 6 above)
Comment (jingyi2811) [inline packages/jellyfish-api-core/src/category/spv.ts:52]: options.feerate = DEFAULT_FEE_RATE_IN_SATOSHI

File: packages/jellyfish-api-core/src/category/spv.ts (hunk 3 above)
Comment (jingyi2811) [reply packages/jellyfish-api-core/src/category/spv.ts:4]: I am not sure why did you you declare this const and export? 
Do you plan to use this somewhere other than current file?

File: website/docs/jellyfish/api/spv.md (hunk 4 above)
Comment (jingyi2811) [inline website/docs/jellyfish/api/spv.md:60]: feeRate

File: website/docs/jellyfish/api/spv.md [hunk 8]
```diff
@@ -50,3 +50,22 @@ interface ReceivedByAddressInfo {
...
+}
+
+interface SendToAddressOptions {
+  feerate?: BigNumber
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/spv.md:64]: feeRate

File: website/docs/jellyfish/api/spv.md [hunk 9]
```diff
@@ -50,3 +50,22 @@ interface ReceivedByAddressInfo {
...
+}
+
+interface SendMessageResult {
+  txid: string
+  sendmessage: string
```
Comment (jingyi2811) [inline website/docs/jellyfish/api/spv.md:69]: sendMessage

File: website/docs/jellyfish/api/spv.md (hunk 4 above)
Comment (jingyi2811) [inline website/docs/jellyfish/api/spv.md:60]: Shouldn't this be 
sendToAddress (address: string, amount: BigNumber, options = { feeRate: new BigNumber('10000') }): Promise<SendMessageResult>

Which is the latest interface you had just changed in your rpc file?

File: website/docs/jellyfish/api/spv.md (hunk 9 above)
Comment (Jouzo) [reply website/docs/jellyfish/api/spv.md:69]: Cannot change it to camelCase as it is define on `ain` side

Comment (jingyi2811) [reply website/docs/jellyfish/api/spv.md:69]: Haha, okay.

**Review by jingyi2811 — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**